### DISCUSSION TITLE
How to modify the model？Please help me, thank you！

### DISCUSSION THREAD

**Discussion by linkin770** [ultralytics/ultralytics#11755] [Q&A]
Today I tried to modify the convolution block in the model. I added indexes in the task.py and conv.py files, but an error occurred:
Traceback (most recent call last):
  File "E:\python387\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "E:\python387\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "E:\python387\Scripts\yolo.exe\__main__.py", line 7, in <module>
  File "O:\ultralytics\ultralytics\cfg\__init__.py", line 555, in entrypoint
    model = YOLO(model, task=task)
  File "O:\ultralytics\ultralytics\models\yolo\model.py", line 23, in __init__
    super().__init__(model=model, task=task, verbose=verbose)
  File "O:\ultralytics\ultralytics\engine\model.py", line 149, in __init__
    self._new(model, task=task, verbose=verbose)
  File "O:\ultralytics\ultralytics\engine\model.py", line 218, in _new
    self.model = (model or self._smart_load("model"))(cfg_dict, verbose=verbose and RANK == -1)  # build model
  File "O:\ultralytics\ultralytics\nn\tasks.py", line 288, in __init__
    self.model, self.save = parse_model(deepcopy(self.yaml), ch=ch, verbose=verbose)  # model, savelist
  File "O:\ultralytics\ultralytics\nn\tasks.py", line 933, in parse_model
    m_ = nn.Sequential(*(m(*args) for _ in range(n))) if n > 1 else m(*args)  # module
TypeError: __init__() takes 8 positional arguments but 9 were given

Please help me, thank you！

**Comment by glenn-jocher** [comment]
Hi there! It looks like you're trying to customize the convolutional layers and facing an argument mismatch error. This often occurs when the constructor of your modified block expects a different number of arguments than you're providing.

Ensure that the number of parameters you pass matches the constructor's expectations in your custom convolution block. Here’s a general example to guide you:

```python
class CustomConv(nn.Module):
    def __init__(self, in_channels, out_channels, kernel_size, stride, padding, ...):
        super(CustomConv, self).__init__()
        # Initialize your convolution here

# Usage
conv_layer = CustomConv(in_channels=32, out_channels=64, kernel_size=3, stride=1, padding=1, ...)
```

Double-check the arguments in the instantiation of your custom class in `conv.py` and `task.py`. Adjust them to ensure they align with what your class constructor requires. If the issue persists, reviewing the specific changes in your implementation might be necessary to provide more targeted advice.

**Reply by linkin770** [reply]
Thanks for your reply, but this doesn't seem to be an issue for me.
I found the answer this morning. The reason is that there is no need to add the index of the new module in `if m in {}` in the tasks.py file.

**Reply by glenn-jocher** [reply]
Great to hear you found the solution! Indeed, you typically don't need to add the index for new modules directly into the `if m in {}` structure in the `tasks.py` file unless there's a specific condition that requires your module to be treated differently during the parsing or initialization process. Always make sure that your new modules integrate seamlessly without causing disruptions within the flow. Happy coding! 😊 If you need further assistance or have more insights to share, feel free to reach out!
